### ISSUE TITLE
There is no way to exclude areas in all iframes based on a selector

### ISSUE THREAD

**Issue by dylanb** [dequelabs/axe-core#130]
for example, if I only want to analyze form fields outside `<form>` elements, then I have no way to specify the exclude in such a way that it will work in a multi-frame document.

**Comment by dsturley** [comment]
Just exclude the frames?

**Comment by dylanb** [comment]
Thing is I want to analyze the content outside all the forms in all the frames

**Comment by dsturley** [comment]
This looks like a bug.  GitHub was inserting a line break in place of the `<form>` reference, which totally changed the meaning of that sentence... I was so confused.  Would [test tests](https://github.com/dsturley/axe-core/commit/19bd7b2c083ec41022f62cd96e4741737122948b) cover your use case?

We may want to refactor (i.e. break) the `Context` parameter to drop support for the `CSSSelectorArray`.  `Context` would support, `include` and `exclude` given the following parameters:

1. The typical accepted parameters (jQuery-like support)
    1. A direct `Element` reference
    1. Array-like structures that references actual DOM nodes (`NodeList`s, `jQuery` objects, etc)
    1. String selector
    
    ```javascript
    axe.a11yCheck({
      include: 'input, textarea, select, etc',
      exclude: jQuery('form') // direct reference, will only exclude forms in current frame
    }, function () {});
    ```

4.  Constructed `CSSSelectorArray` object.  It would be constructed by calling a utility.  This would allow us to remove the ambiguity of "Is this object not an Array, but Array-like, but also not string but not an Array that contains an Array of strings?".
    
    ```javascript
    axe.a11yCheck({
      include: 'input, select, textarea, etc',
      exclude: axe.util.elementLocatorThingy([
        'iframe.crazy',
        'iframe.long',
        'iframe.selector',
        'div.thingy'])
    }, function () {});
    ```

**Comment by dsturley** [comment]
Another problem is that currently including specific areas exclude iframes that are outside those areas.  The `input, select, textarea` example highlights this nicely.  A user might be expecting to see all form inputs in all frames, but will only get them in the top frame.

**Comment by dylanb** [comment]
I don't like totally blowing away the current specific selector for the following reasons:

1) totally breaks the current API, and
2) means that we cannot pass an address from one of our issues back into the API

We could fix #2 by making our addresses this new type of object but that breaks the API even more.

Why don't we just add a simple string selector to include and exclude that would always be interpreted as being executed in all frames.

We could mix this with selector arrays to achieve any targeting we want.

Example: analyze only in my main content area inside all `<form>` elements (in all frames)

```
{
    include: [
        ["[role='main']"],
        "form"
    ]
}
```

Example: Analyze in all `<form>` elements (in all frames) but not inside the irritating survey popup thingy

```
{
    include: ["form"],
    exclude: [["iframe.foresee"]]
} 
```

Example: Analyze everything in all frames except the reviews from that third party review provider (in all frames)

```
{
    exclude: [".bv-extensions-panel"]
}
```

**Comment by WilcoFiers** [comment]
3 years since this issue was opened, and I have yet to see a single use case for it. I'm going to assume this isn't a problem for anyone and close the issue.
